pkg/rtc/transportmanager.go

```diff
@@ -609,3 +623,32 @@ func (t *TransportManager) ProcessPendingPublisherDataChannels() {
 		}
 	}
 }
+
+func (t *TransportManager) OnReceiverReport(dt *sfu.DownTrack, report *rtcp.ReceiverReport) {
+	t.mediaLossProxy.HandleMaxLossFeedback(dt, report)
+}
+
+func (t *TransportManager) onMediaLossUpdate(loss uint8) {
+	t.udpLossUnstableCount <<= 1
+	if loss >= uint8(255*udpLossFracUnstable/100) {
+		t.udpLossUnstableCount |= 1
+		if bits.OnesCount32(t.udpLossUnstableCount) >= udpLossUnstableCountThreshold {
+			if t.udpRTT > 0 && t.tcpRTT < uint32(float32(t.udpRTT)*1.3) && t.tcpRTT < tcpGoodRTT {
+				t.udpLossUnstableCount = 0
+				t.params.Logger.Debugw("udp connection unstable, switch to tcp")
+				t.handleConnectionFailed(true)
+				if t.onAnyTransportFailed != nil {
+					t.onAnyTransportFailed()
+				}
+			}
+		}
+	}
```

Comment:
Should this function actually check that the transport is UDP? Maybe setup the `MediaLossProxy` callback only after the `SUBSCRIBER` ICE connection is established.

Reply:
Okay. I was more thinking about the case of `preference = None`, but ICE chooses a different path. With TCP, there would be no loss. So, that should be okay. For TURN/TLS also, loss should be low even though the path between TURN server and SFU will be UDP.